### ISSUE TITLE
The Tetra-X kickset incorrectly defines at least one T-Spin Mini scenario as a full T-Spin

### ISSUE THREAD

**Issue by ZaptorZap** [tetrio/issues#626]
### Checklist 
- [x] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this issue is not already reported.
- [x] I am using the **newest version** of TETR.IO.
- [x] I know **how to reproduce the issue**.
- [x] I am using a **supported browser** or the Desktop Client.
- [x] I am not using anything that **modifies game code** (e.g. TETR.IO+).
- [x] I am certain this is a **bug**, not a feature request.
### Describe the bug/issue
[The East to North kick #2 for the T piece in SRS(+)](https://four.lol/srs/t-kicks#kick-2-2), or the West to North of this same kick causes a T-Spin mini, as it should, assuming TETR.IO uses 3 point detection; with default settings. This behavior is currently implemented, and is included in the example-replays.zip file further below in example 1.

This same kick is included in the Tetra-X kickset, and after reproducing it, you will end up with a full T-Spin as opposed to a T-Spin mini. This is demonstrated in example 2.

At first, I believed this was a nod to Tetra Legends's ingame "TETRA-X" mode, which uses immobile spin detection, as immobile spin detection does not include specifications for mini T-Spins. This is not the case, as T-Spin minis utilizing [North to West kick 1](https://ch.tetr.io/api/users/zaptorzap)/North to East kick 1 will count for a T-Spin mini under Tetra-X(shown in example 3!). It doesn't take a genius to try something that would count for immobile systems and not 3 point detection spins, as example 4 shows a spin that counts in Tetra Legends and not in TETR.IO.

Here's a .zip file containing examples 1, 2, 3, and 4: 
[issue 626 TTR examples.zip](https://github.com/tetrio/issues/files/5962808/issue.626.TTR.examples.zip)
### Reproducing
1. Start either a solo custom game or an online multiplayer lobby with the "TETRA-X" kickset enabled and "ALLOWED SPINS" set to its default "T-SPINS"
2. Create a spin that resembles [the East to North kick #2 for the T piece in SRS](https://four.lol/srs/t-kicks#kick-2-2)
3. Note that this move causes an attack of 2 lines(+B2B) or creates a score of 800 points(*b2b) and causes the action text of "T-SPIN SINGLE" as opposed to "T-SPIN MINI SINGLE"
4. If you do this on a solo custom game, you can also confirm this with the "FULL" stats in the after-action stat screen.
### Expected behavior
The T-Spin should count as a T-Spin mini, considering in other kicksets, such as ASC, ARS, or SRS+, it counts as a T-Spin mini. 
### Screenshots
with SRS+:
![image](https://user-images.githubusercontent.com/64891705/107594720-76291580-6bc7-11eb-8e33-6e0c53e4c61f.png)
with Tetra-X
![image](https://user-images.githubusercontent.com/64891705/107594459-d10e3d00-6bc6-11eb-9e08-75da86408e59.png)
### Your setup
 - **OS and version**: Linux, Debian Buster (10) x86_64
 - **Browser and version**: Desktop V7, 6.0.0-Production
### Additional context
I mentioned ARS and ASC up there and it got me thinking; perhaps you could embrace this issue and make the T piece count immobile spins when used with the ASC and Tetra-X kicktables. The _Tetris The Grand Master_ series also has some seriously dumb rules for what counts as a T-Spin as well. SRS, SRS-X, and SRS+ could also make a switch to the guideline default 4-point detection for all pieces instead of the hybrid "immobile for every piece except T" option we have at the moment. In a way, this is related to #622.

**Comment by UU6202** [comment]
I can't seem to find information about "guideline default 4-point detection for all pieces" but only know it is implemented in Nullpomino from [this forum](https://harddrop.com/forums/index.php?showtopic=5492).

**Comment by ZaptorZap** [comment]
It's also implemented in the Tetris Party Deluxe series by Hudson Soft (Tetris Party, Tetris Party Deluxe, and Tetris Axis)
an example of this [posted by user Ðimentio#0001 on the TETR.IO Discord server:](https://discord.com/channels/673303546107658242/673303546564968566/772259603115278367)

https://user-images.githubusercontent.com/64891705/109087898-35e78e00-76c3-11eb-8569-9d22cf4821f5.mp4

(note the mini spins in this video; mini spins **cannot** occur with immobile spin detection, the spin detection TETR.IO uses for allspin (aside from T, which is still handled with 4 point detection regardless))

**Comment by ZaptorZap** [comment]
**Yep, this still happens:**

https://user-images.githubusercontent.com/64891705/119592963-4379e600-bd8e-11eb-9ac8-67e63c1fb3a4.mp4

**Comment by deedeeen** [comment]
@ZaptorZap, maybe recreate the "bug" in https://tetralegends.app?

**Comment by ZaptorZap** [comment]
Yes, [I'm very well aware]() of Tetra Legends's immobile detection:
> At first, I believed this was a nod to Tetra Legends's ingame "TETRA-X" mode, which uses immobile spin detection, as immobile spin detection does not include specifications for mini T-Spins. This is not the case, as T-Spin minis utilizing [North to West kick 1](https://ch.tetr.io/api/users/zaptorzap)/North to East kick 1 will count for a T-Spin mini under Tetra-X(shown in example 3!). It doesn't take a genius to try something that would count for immobile systems and not 3 point detection spins, as example 4 shows a spin that counts in Tetra Legends and not in TETR.IO.

**Comment by deedeeen** [comment]
Should have reread the issue - deleted comment.

**Comment by ZaptorZap** [comment]
don't worry about it lol

**Comment by deedeeen** [comment]
lol now we wait to challenge osk for the implementation

**Comment by deedeeen** [comment]
just something - is it a good idea to set graphics to Ultra _and_ play under fast speed to show the problem? by the nature of the bug, none of them are necessary

**Comment by ZaptorZap** [comment]
It was a rather quickly made video for a bug report that's already quite detailed. If you'd like slower examples, please pick up [a copy of my `.zip` archive which contains replays you can slow down yourself to a quarter of its speed.](https://github.com/tetrio/issues/files/5962808/issue.626.TTR.examples.zip)

**Comment by deedeeen** [comment]
What I mean here is, it is quite unnecessary to formally _record_ them with the high settings. I can personally follow the inputs, to clarify.

**Comment by ZaptorZap** [comment]
This is all just a byproduct of my natural setting selections. You said it yourself, it's not necessary to use these high settings to record. It is not necessary either, however, to lower the settings to cause the bug described as it is a bug with the engine, not with the renderer.

**Comment by deedeeen** [comment]
It is a bug with the engine; as such I have based my suggestion to reduce both gameplay speed and graphics on accessibility. After all, you're here to report a bug, not show off any of them. So in my opinion, it's best if everything possible is minimal.

**Comment by ZaptorZap** [comment]
So sorry for the wait, but [here's **an improved version**](https://www.youtube.com/watch?v=Xb3KM98Jwkk) of [my previous video.](https://github.com/tetrio/issues/issues/626#issuecomment-848405807) Feel free to send your friends a copy! I hope it lets them understand the previously uploaded *extremely* high-speed footage.

**Comment by deedeeen** [comment]
A good one - it could have some room for improvement, such as a higher locking delay. 
_note:_ It is usually not recommended due to its gameplay alteration but this is an exception.

**Comment by ZaptorZap** [comment]
yop still exists

**Comment by KonSola5** [comment]
This may be intended, as the kick used was the 5th kick in the Tetra-X kick table (#310).
![explanation](https://github.com/tetrio/issues/assets/125081901/f1d253fd-4cd9-47df-862c-6e975e1ecd63)

**Comment by ZaptorZap** [comment]
Well that's a good explanation, but I still believe this to be unintended behavior. Afterall, the original system has no concept of Mini T-Spins.

**Comment by KonSola5** [comment]
In ASC, the aforementioned kick is the 4th kick in the kick table (3rd not counting the natural rotation), that's why it's Mini.
![](https://github.com/tetrio/issues/assets/125081901/3ebe161c-a716-4360-b353-1cb7dd49f0b5)

ASC also uses immobile spin detection in its home game, just like Tetra-X.

**Comment by ZaptorZap** [comment]
Sure, but these are SRS standards. It's hard to take a 2D array and know immediately if something like this is gonna happen. That's why I presume this was an oversight, not an intentional design choice. (The intentional choice, if there were one, would be to make all T-Spins immobile, just like in the source game.)
